### DISCUSSION TITLE
How to export a model (from e.g., Blender) to a finite difference grid?

### DISCUSSION THREAD

**Discussion by laura-HeroicaTech** [community/community#143140] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hoping we can get some help on this. We want to do electromagnetic simulations and need a regular finite difference mesh to do so. We want to be able to export a model (from e.g., Blender) to a finite difference grid and specify the start, stop, and step of each x, y, and z axis.

We tried this with Blender using the re-mesh > blocks modifier. It looks promising, but it does not yield a dense mesh as needed for EM simulations.

We are open to paying for tools, but the ~$15k cost of the big names (Altair, etc.) are out of our reach at the moment.

Thank you in advance for any help or insights!

**Comment by SyedAliJunaidSherazi** [comment]
Hi there,

Exporting a model from Blender to a finite difference grid for electromagnetic simulations can be a bit challenging, but there are a few approaches you can try:

1. **Use the Remesh Modifier**: As you've mentioned, the Remesh > Blocks modifier is a good start. To achieve a denser mesh, you might want to experiment with different voxel sizes or settings. Ensure that the settings are optimized for the level of detail you need.

2. **Grid Generation Add-ons**: Consider using Blender add-ons that can help generate grids or meshes more suitable for simulations. Add-ons like **BlenderGIS** or **MeshTools** can assist with creating regular grids, although they may require some customization for your specific needs.

3. **Exporting to a Compatible Format**: After creating a mesh, export it in a format that your simulation software can read (like STL, OBJ, or other supported formats). Ensure that your export settings preserve the mesh density and structure.

4. **Mesh Processing Tools**: Look into standalone mesh processing tools like **GMSH** or **FreeCAD**. They can create finite element or finite difference meshes and may allow for specifying parameters like start, stop, and step for each axis.

5. **Scripting in Blender**: If you have programming skills, you can create a script in Blender using Python to generate a grid directly according to your specifications. This can give you more control over the mesh density and structure.

6. **Budget-Friendly Software Alternatives**: Since high-cost options are out of reach, consider exploring other simulation tools that may offer lower costs or free versions, such as **OpenFOAM** or **Elmer FEM**. They might have the capabilities you need for EM simulations.

7. **Community Forums**: Engage with communities or forums focused on electromagnetic simulations or Blender. They may have shared tools or methods that can help you achieve a suitable mesh.

I hope these suggestions help you find a suitable solution! Good luck with your simulations, and feel free to ask if you have any further questions :)

**Comment by divsansi** [comment]
Response #10108
